### ISSUE TITLE
Display update dialog even when the latest version of the app is installed in iOS

### ISSUE THREAD

**Issue by rohnsht** [larryaasen/upgrader#113]
The package display update available popup every the app is opened, even when the latest version of the app is installed in iOS. Also, it keeps displaying the popup even  after `Ignore` is pressed.

**Comment by larryaasen** [comment]
Can you include the upgrade console log?

**Comment by rohnsht** [comment]
Here is the console log:
```
flutter: FirebaseToken: dD43rUT8BkW0gsOn046J6W:APA91bGwMEnSGfLnpDQjGKXPIl0snfNaLaN_YE00e-t5hKW8Nq5UkesIQUgbDw3ebD2KkJWSgs-BljTbL3Oyuc8iih0k2xp5H5trJlKJ8ODod2IQhL_DcUPoWT-54GIi9z2l2dhyTr6i
flutter: upgrader: package info packageName: com.ego.myQv
flutter: upgrader: package info appName:
flutter: upgrader: package info version: 1.1.1
flutter: upgrader: countryCode: AU
flutter: {status: -5, message: invalid token}
flutter: invalid token
flutter: {status: -5, message: invalid token}
4
flutter: invalid token
flutter: upgrader: blocked: false
flutter: upgrader: debugDisplayAlways: false
flutter: upgrader: debugDisplayOnce: true
flutter: upgrader: hasAlerted: false
flutter: upgrader: shouldDisplayUpgrade: true
flutter: upgrader: shouldDisplayReleaseNotes: true
flutter: upgrader: showDialog title: Update App?
flutter: upgrader: showDialog message: A new version of  is available! Version 1.1.1 is now available-you have 1.1.1.
flutter: upgrader: showDialog releaseNotes: 1. Fix font scaling issue.


```

**Comment by larryaasen** [comment]
Maybe you should try: debugDisplayOnce: false

**Comment by eldadj** [comment]
I have also stumbled on this issue which was very strange. Doing a curl to the update url for iTunes, returned the correct info, but a debug of itunes_search_api.dart was returning a wrong information. Further debugging indicated that the http get on line
` file: itunes_search_api.dart
42: final response = await client!.get(Uri.parse(url));
`
indicated that the http client was returning a cached response. A check of the cache header showed `cache-max-age of: 30580`

What I did was add a cache-buster to the url on line 36
`final url = '${lookupURLByBundleId(bundleId, country: country ??= '')!}&_cb=${DateTime.now().microsecondsSinceEpoch.toString()}';`

**Comment by sjrphx** [comment]
I'm also experiencing this same issue. Only on iOS.

**Comment by larryaasen** [comment]
@eldadj I appreciate the cache buster idea. I can incorporate that into the next release.

**Comment by larryaasen** [comment]
I have published version 3.5.0 to use a cache buster along with other updates. Thanks to @eldadj for the discovery and work on this issue.
